lib/nostrum/struct/channel.ex

```diff
@@ -37,132 +135,282 @@ defmodule Nostrum.Struct.Channel do
-    def to_string(channel), do: @for.mention(channel)
-  end
-
-  @typedoc "The channel's id"
+  @typedoc """
+  The id of the channel object.
+  """
   @type id :: Snowflake.t()
 
-  @typedoc "The id of the channel's guild"
+  @typedoc """
+  The type of channel.
+
+  |      | Channel Type         |                                                                                        |
+  | ---- | -------------------- | -------------------------------------------------------------------------------------- |
+  | `0`  | Guild Text           | `GUILD_TEXT`            _A text channel within a server_                               |
+  | `1`  | Direct Message       | `DM`                    _A direct message between users_                               |
+  | `2`  | Guild Voice          | `GUILD_VOICE`           _A voice channel within a server_                              |
+  | `3`  | Group Direct Message | `GROUP_DM`              _A direct message between multiple users_                      |
+  | `4`  | Guild Category       | `GUILD_CATEGORY`        _A category that contains up to 50 channels_                   |
+  | `5`  | Guild Announcements  | `GUILD_NEWS`            _A channel that users can follow and crosspost from_           |
+  | `6`  | Guild Store          | `GUILD_STORE`           _A channel in which developers can sell their game on Discord_ |
+  | `10` | Guild News Thread    | `GUILD_NEWS_THREAD`     _A temporary sub-channel within a `5` channel_                 |
+  | `11` | Guild Public Thread  | `GUILD_PUBLIC_THREAD`   _A temporary sub-channel within an `0` channel_                |
+  | `12` | Guild Private Thread | `GUILD_PRIVATE_THREAD`  _A temporary private sub-channel within an `0`                 |
+  | `13` | Guild Stage Voice    | `GUILD_STAGE_VOICE`     _A voice channel for hosting events with an audience_          |
+
+  More information about _Discord Channel Types_ can be found on the [Discord API Channel Type Documentation](https://discord.com/developers/docs/resources/channel#channel-object-channel-types).
+  """
+  ## Unsure if it should be here or at the top of the documentation, at both for now.
+  @type type :: integer()
+
+  @typedoc """
+  The id of the guild the channel is located in.
+  """
   @type guild_id :: Guild.id()
 
-  @typedoc "The ordered position of the channel"
+  @typedoc """
+  The position of the channel in the sidebar of the guild.
+  """
   @type position :: integer()
 
-  @typedoc "The list of overwrites"
+  @typedoc """
+  A list of permission overwrites applied to the channel.
+  """
   @type permission_overwrites :: [Overwrite.t()]
 
-  @typedoc "The name of the channel"
+  @typedoc """
+  The name of the channel.
+  """
   @type name :: String.t()
 
-  @typedoc "Current channel topic"
+  @typedoc """
+  The topic of the channel.
+  """
   @type topic :: String.t()
 
-  @typedoc "If the channel is nsfw"
-  @type nsfw :: boolean
+  @typedoc """
+  The channels NSFW setting.
```

Comment:
I think it would be clearer to keep the wording on this indicate that it's a boolean field, because "setting" may sound like there's multiple values to be set here:

```suggestion
  Whether the NSFW setting is enabled for this channel.
```